### ISSUE TITLE
enhance jobwait() (WAS: E5677 with system('$(</dev/stdin) ...')

### ISSUE THREAD

**Issue by haynesgt** [neovim/neovim#5732]
- `nvim --version`: NVIM v0.2.0-54-g8da23cb
- Vim (version: VIM - Vi IMproved 7.4 (2013 Aug 10, compiled Nov 24 2016 16:42:52)) behaves differently? YES
- Operating system/version: Ubuntu 14.04.5 / Linux 3.13.0-96-generic #143-Ubuntu SMP Mon Aug 29 20:15:20 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
- Terminal name/version: GNU bash, version 4.3.11(1)-release (x86_64-pc-linux-gnu)
- `$TERM`: xterm

### Actual behaviour

E5677: Error writing input to shell-command: EPIPE

### Expected behaviour

No error / input from help page should be assigned to the variable

### Steps to reproduce using `nvim -u NORC`

```
nvim -u NORC
:help set
:call system('BUFF=$(</dev/stdin)', join(getline(1, '$'), '\n'))
```

The `:call` command works when not sent from a help page.

**Comment by justinmk** [comment]
This is because the process closes before we can write to it. Even in Vim, what would this accomplish? `$BUFF` needs to be written somewhere before the shell exits. And you can already write buffer contents via

    :%write foo

`jobstart()` and `jobwait()` can be combined to achieve pretty much anything else.

tl;dr: This is a known limitation that we can't change because internally we don't use shell-specific tricks to work with processes, we work with processes directly (via libuv). The details are in https://github.com/neovim/neovim/pull/5502.

A way to reduce this impact of this quirk is to enhance `jobwait()` to be a bit easier to use.

**Comment by haynesgt** [comment]
Regarding what this accomplishes, this is one way to multiplex the buffer contents within a shell command. 

In particular, I have a function that saves a backup of the buffer:
`call system('mkdir -p /tmp/vimback && BUFF=$(</dev/stdin) && HASH=$(echo "$BUFF" | sha256sum | head -c64) && echo "$BUFF" | gzip > "/tmp/vimback/$HASH.gz" && echo $(date "+%Y-%m-%d %H:%m:%S") "/tmp/vimback/$HASH.gz" "' . expand("%") . '" >> /tmp/vimback/index', join(getline(1, '$'), "\n"))`
This saves buffers to /tmp/vimback/<buffer hash>.gz and keeps an index file in /tmp/vimback/index

Using `"%w !BUFF=$(</dev/stdin); ... $BUFF ..."` does not work because it removes the newlines from BUFF

Thanks
